stbt.ocr: Add text_color_threshold parameter

drothlis opened the pull request:
~~I've wanted this in the past, though now I don't remember the exact use case so I don't have any images to use for a self-test.~~

I have added a testcase (see [comment](#issuecomment-378564281) below).

wmanley commented on stbt/__init__.py:
```diff
@@ -408,7 +408,9 @@ def ocr(frame=None, region=Region.ALL,
-        for example for white text on a light-colored background.
+        for example for white text on a light-colored background. The threshold
+        can be adjusted further by passing a 4th number in the tuple, also
+        between 0 and 255.
```
I'm not sure of this as an API.  Perhaps a `text_color_threshold` would be better.  I've been thinking of adding a `Color` class much like we have `Region` and I don't see how it would fit in here.

wmanley commented:
> I don't remember the exact use case so I don't have any images to use for a self-test.

I think we'll have to wait until we've got some examples where this helps before merging it.

drothlis commented:
I've rebased onto master and added a testcase that currently fails at the default text_color threshold of 25.

Source image:
![blue-search-white-guide](https://user-images.githubusercontent.com/12424/38304273-b57b5840-3800-11e8-8f2b-9161bb9e045f.png)

text_color | threshold=25 | threshold=50
--- | --- | ---
(200,200,200) | ![200-25](https://user-images.githubusercontent.com/12424/38304209-85ddc302-3800-11e8-9880-5b6e29894c9e.png) Reads `'GUidG'` |
(220,220,220) | ![220-25](https://user-images.githubusercontent.com/12424/38304210-86180dc8-3800-11e8-85bf-9bbbdb78e5e5.png) Reads `'GUIde'` | ![220-50](https://user-images.githubusercontent.com/12424/38304212-8643783c-3800-11e8-9bf5-f385cba1de5f.png) Reads `'Guide'`
(240,240,240) | ![240-25](https://user-images.githubusercontent.com/12424/38304213-8668d118-3800-11e8-9969-e8a8b192b2de.png) Reads `"(Lamb '*"` |

drothlis commented:
Rebased onto master, and changed to use a separate `text_color_threshold` parameter instead of shoving the threshold into the `text_color` tuple.

wmanley commented on _stbt/core.py:
```diff
@@ -1220,7 +1220,8 @@ def wait_for_motion(
             lang=None, tesseract_config=None, tesseract_user_words=None,
-            tesseract_user_patterns=None, upsample=True, text_color=None):
+            tesseract_user_patterns=None, upsample=True, text_color=None,
+            text_color_threshold=25):
```
If we're thinking that a different default might be a good idea perhaps this should be `=None` and we can look up the default with `get_config`?

drothlis replied:
Good idea.
